presto-main/src/main/java/io/prestosql/operator/window/ReflectionWindowFunctionSupplier.java

```diff
@@ -39,11 +41,20 @@ public ReflectionWindowFunctionSupplier(String name, Type returnType, List<? ext
     public ReflectionWindowFunctionSupplier(Signature signature, Class<T> type)
     {
         super(signature, getDescription(requireNonNull(type, "type is null")));
+        canIgnoreNulls = ValueWindowFunction.class.isAssignableFrom(type);
```

Comment:
Setting this here unconditionally isn't correct for the fallback case. The variable needs to reflect which version of the constructor is actually available.

How about adding a `ConstructorType` enum? Then the `newWindowFunction` method would be more direct and wouldn't need to re-check the signature:
```java
switch (constructorType) {
    case NO_INPUTS:
        return constructor.newInstance();
    case INPUTS:
        return constructor.newInstance(inputs);
    case INPUTS_IGNORE_NULLS:
        return constructor.newInstance(inputs, ignoreNulls);
}